File: apollo-api/src/main/java/com/apollographql/apollo/api/FragmentResponseFieldMapper.java
package com.apollographql.apollo.api;

import java.io.IOException;

/**
 * Maps the response back to the data for a fragment
 * @param <T>
 */
public interface FragmentResponseFieldMapper<T> {
  T map(final ResponseReader responseReader, String conditionalType) throws IOException;
}

File: apollo-api/src/main/java/com/apollographql/apollo/api/FragmentResponseFieldMapper.kt
package com.apollographql.apollo.api

import java.io.IOException

/**
 * FragmentResponseFieldMapper is responsible for mapping the response back to a fragment of type T.
 */
interface FragmentResponseFieldMapper<T> {
  @Throws(IOException::class)
  fun map(responseReader: ResponseReader, conditionalType: String): T
}
